### ISSUE TITLE
[a11y]: Search [expandable] - collapse status not announced

### ISSUE THREAD

**Issue by andreancardona** [carbon-design-system/carbon#14817]
### Package

@carbon/react

### Browser

FF, Safari & iPhone

### Package version

@carbon/react@1.32.0

### Automated testing tool and ruleset

Manual QA

### Assistive technology

JAWS & VO

### Description
Collapse Status not announced when the component loads but when we click on Search Button Expand status announced. 

If the screen reader does not announce the status change in a page element, it’s difficult for users to understand it on their own. This defect mainly impacts users who are blind. 

### Recommended Fix
Both Expand and Collapse status needs to be announced correctly when we navigate using the screen reader

### WCAG 2.1 Violation

4.1.2 Name, Role, Value
